### PR TITLE
update curve25519-dalek to 4.0.0-rc.3; bump version to 4.0.1

### PR DESCRIPTION
**Pull request by conradoplg**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name = "ed25519-zebra"
 # Before publishing:
 # - update CHANGELOG.md
-version = "4.0.0"
+version = "4.0.1"
 rust-version = "1.65.0"
 authors = ["Henry de Valence <hdevalence@hdevalence.ca>"]
```
Comment (teor2345) [inline Cargo.toml:7]: I think we missed adding ZF as an author here a long time ago.

```suggestion
authors = ["Henry de Valence <hdevalence@hdevalence.ca>", "Zcash Foundation <zebra@zfnd.org>"]
```

Comment (teor2345) [reply Cargo.toml:7]: Our copyright is in the license files, so there's no legal impact here. But it helps to let people know.

**Review by teor2345 — APPROVED** [review]
Thanks, happy to approve without the author changes if this is urgent.

**Comment by conradoplg** [comment]
@teor2345 thank for pointing that out! Needs reapproval

**Review by teor2345 — APPROVED** [review]

**Comment by pinkforest** [comment]
psst 4.0.0 has landed :tada:

**Pull request merged**